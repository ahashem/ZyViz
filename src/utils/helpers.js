/**
 * Provide shortened version (3 letters) of the provided Day of Week
 * @param weekDay
 * @return {string} shortened weekday
 */
const shortenWeekDay = (weekDay) => {
  const weekDays = {
    'Mondays': 'Mon',
    'Monday': 'Mon',
    'Tuesdays': 'Tue',
    'Tuesday': 'Tue',
    'Wednesdays': 'Wed',
    'Wednesday': 'Wed',
    'Thursdays': 'Thu',
    'Thursday': 'Thu',
    'Fridays': 'Fri',
    'Friday': 'Fri',
    'Saturdays': 'Sat',
    'Saturday': 'Sat',
    'Sundays': 'Sun',
    'Sunday': 'Sun',
  };

  return weekDays[`${weekDay}`];
};

/**
 * Format currency with USD
 * @param amount
 * @return {string}
 */
const currencyTickFormat = (amount) => (((amount / 1000) >= 1) ? `$${(amount / 1000)}K` : `$${amount}`);

export { shortenWeekDay, currencyTickFormat };
