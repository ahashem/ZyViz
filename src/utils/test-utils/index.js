import crossfilter from '../../components/crossfilter/crossfilter';
import NormalizeData from '../../views/CrossVisualizer/visualizer-data-structure/data-normalizer';
import mockOrdersJSON from './mock-test-orders';
import GenerateDimensions from '../../views/CrossVisualizer/visualizer-data-structure/dimensions-manager';

const mockOrdersCrossfilter = crossfilter(NormalizeData(mockOrdersJSON));
const mockOrdersDimensions = GenerateDimensions(mockOrdersCrossfilter);

export { mockOrdersJSON as jsonMock, mockOrdersCrossfilter as crossFilterMock, mockOrdersDimensions as dimensionsMock };
