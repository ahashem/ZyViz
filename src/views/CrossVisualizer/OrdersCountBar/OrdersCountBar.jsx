import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FlexGridRow from '../../../components/flex-wrapper/FlexGridRow';
import BarChart from '../../../components/BarChart/BarChart';
import { Col } from 'antd';

/**
 * @component OrdersCountBar
 */
class OrdersCountBar extends Component {
  static propTypes = {
    orders: PropTypes.shape({}),
    dimension: PropTypes.shape({}),
  };

  static defaultProps = {
    orders: {},
    dimension: {},
  };

  /**
   * Receive a crossfilter dimensions, create groups to use for charts
   * @param {Object} dimensions
   * @return {Object<Array<crossfilter.Grouping>>}
   */
  prepareOrdersCountData = (dimensions) => {
    // Serialize and group data using crossfilter to be
    // for-example: {key: 'paymentMethod', value: 20}

    // Orders by Branch
    const ordersByBranch = dimensions.ordersByBranch;
    const ordersByBranchData = ordersByBranch.group().reduceCount().all();

    // Orders by DeliveryArea
    const ordersByDeliveryArea = dimensions.ordersByDeliveryArea;
    const ordersByDeliveryAreaData = ordersByDeliveryArea.group().reduceCount().top(20);

    // Orders by Order Day of Week
    // Week days (Saturday - Sunday ...etc)
    const ordersByWeekDay = dimensions.ordersByWeekDay;
    const ordersByWeekDayData = ordersByWeekDay.group().reduceCount().all();

    return {
      ordersByBranchData,
      ordersByDeliveryAreaData,
      ordersByWeekDayData,
    };

  };

  /**
   * Uplift filter triggers
   * @param {*|string} dimension
   * @param {Number} selectedKey
   * @param {Boolean} deselected
   */
  crossFilterSelected = (dimension, selectedKey, deselected = false) => {
    const { onFilter } = this.props;
    onFilter(dimension, selectedKey, deselected);
  };


  render() {
    const { dimensions } = this.props;
    const ordersCrossed = dimensions ? this.prepareOrdersCountData(dimensions) : null;

    return (
      <div>
        <FlexGridRow justify="space-around" gutter={8}>

          <Col
            xs={24}
            sm={24}
            md={8}
            lg={8}
          >
            <BarChart
              height={250}
              width={350}
              data={ordersCrossed && ordersCrossed.ordersByBranchData}
              onClick={(selectedIndex, deselected) => this.crossFilterSelected(
                'ordersByBranch',
                ordersCrossed.ordersByBranchData[selectedIndex].key,
                deselected
              )}
              x="key"
              y="value"
            />
            <BarChart
              height={250}
              width={350}
              data={ordersCrossed && ordersCrossed.ordersByWeekDayData}
              onClick={(selectedIndex, deselected) => this.crossFilterSelected(
                'ordersByWeekDay',
                ordersCrossed.ordersByWeekDayData[selectedIndex].key,
                deselected
              )}
              x="key"
              y="value"
            />
          </Col>

          <BarChart
            height={650}
            width={720}
            data={ordersCrossed && ordersCrossed.ordersByDeliveryAreaData}
            onClick={(selectedIndex, deselected) => this.crossFilterSelected(
              'ordersByDeliveryArea',
              ordersCrossed.ordersByDeliveryAreaData[selectedIndex].key,
              deselected
            )}
            x="key"
            y="value"
          />

        </FlexGridRow>
      </div>
    );
  }
}

export default OrdersCountBar;
