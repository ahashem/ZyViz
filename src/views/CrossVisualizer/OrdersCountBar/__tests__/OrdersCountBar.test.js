import React from 'react';
import { mount, shallow } from 'enzyme';

import OrdersCountBar from '../OrdersCountBar';
import { crossFilterMock, dimensionsMock } from '../../../../utils/test-utils';

describe('<OrdersCountBar /> Component', () => {
  let rendered, onFilter;

  beforeEach(() => {
    const orders = crossFilterMock;
    const dimensions = dimensionsMock;
    onFilter = jest.fn();

    rendered = shallow(<OrdersCountBar
      orders={orders}
      dimensions={dimensions}
      onFilter={onFilter}
    />);
  });

  it('should render children and match snapshot', () => {
    expect(rendered).toMatchSnapshot();
  });
});

describe('<OrdersCountBar /> Inner Components', () => {
  let rendered, onFilter;

  beforeEach(() => {
    const orders = crossFilterMock;
    const dimensions = dimensionsMock;
    onFilter = jest.fn();

    rendered = mount(<OrdersCountBar
      orders={orders}
      dimensions={dimensions}
      onFilter={onFilter}
    />);
  });

  it('should filter data by clicking on the chart', () => {
    expect(rendered.find('VictoryBar')).toHaveLength(3);
    expect(rendered.find('BarChart')).toHaveLength(3);
  });

  it('should filter data by clicking on the chart', () => {
    rendered.instance().crossFilterSelected();
    expect(onFilter).toBeCalled();
  });

  it.skip('should filter data by clicking on the chart', () => {
    const mockFunc = jest.fn();
    rendered.instance().crossFilterSelected = () => jest.fn();
    rendered.find('BarChart').forEach((node) => {
      node.simulate('click');
      expect(mockFunc).toBeCalled();
    });

  });
});
