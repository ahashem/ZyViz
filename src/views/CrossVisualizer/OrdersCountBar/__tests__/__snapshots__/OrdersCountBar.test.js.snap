// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrdersCountBar /> Component should render children and match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrdersCountBar
    dimension={Object {}}
    dimensions={
      Object {
        "ordersByBranch": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
        "ordersByDate": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
        "ordersByDeliveryArea": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
        "ordersByPaymentMethod": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
        "ordersBySize": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
        "ordersByTimeOfDay": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
        "ordersByWeekDay": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
      }
    }
    onFilter={[MockFunction]}
    orders={
      Object {
        "add": [Function],
        "all": [Function],
        "allFiltered": [Function],
        "dimension": [Function],
        "groupAll": [Function],
        "isElementFiltered": [Function],
        "onChange": [Function],
        "remove": [Function],
        "size": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <FlexGridRow
        gutter={48}
        justify="space-around"
      >
        <BarChart
          data={
            Array [
              Object {
                "key": "Branch A",
                "value": 4,
              },
              Object {
                "key": "Branch B",
                "value": 4,
              },
              Object {
                "key": "Branch C",
                "value": 3,
              },
              Object {
                "key": "Branch D",
                "value": 6,
              },
              Object {
                "key": "Branch E",
                "value": 3,
              },
              Object {
                "key": "Branch F",
                "value": 7,
              },
            ]
          }
          onClick={[Function]}
          x="key"
          y="value"
        />
        <BarChart
          data={
            Array [
              Object {
                "key": "Rai",
                "value": 2,
              },
              Object {
                "key": "Reggai",
                "value": 2,
              },
              Object {
                "key": "Zahra",
                "value": 2,
              },
              Object {
                "key": "Qibla",
                "value": 2,
              },
              Object {
                "key": "Doha",
                "value": 1,
              },
              Object {
                "key": "Fahaheel",
                "value": 1,
              },
              Object {
                "key": "Abu Halifa",
                "value": 1,
              },
              Object {
                "key": "Jaber Al Ahmad",
                "value": 1,
              },
              Object {
                "key": "Al Qusour",
                "value": 1,
              },
              Object {
                "key": "Jaber Al Ali",
                "value": 1,
              },
              Object {
                "key": "Mirqab",
                "value": 1,
              },
              Object {
                "key": "Nahda",
                "value": 1,
              },
              Object {
                "key": "Nuzha",
                "value": 1,
              },
              Object {
                "key": "Qasser",
                "value": 1,
              },
              Object {
                "key": "Rabiya",
                "value": 1,
              },
              Object {
                "key": "Hadiya",
                "value": 1,
              },
              Object {
                "key": "Qortuba",
                "value": 1,
              },
              Object {
                "key": "Faiha",
                "value": 1,
              },
              Object {
                "key": "Al Bidae",
                "value": 1,
              },
              Object {
                "key": "Abu Al Hasaniya",
                "value": 1,
              },
            ]
          }
          onClick={[Function]}
          x="key"
          y="value"
        />
        <BarChart
          data={
            Array [
              Object {
                "key": "Fridays",
                "value": 1,
              },
              Object {
                "key": "Mondays",
                "value": 1,
              },
              Object {
                "key": "Saturdays",
                "value": 6,
              },
              Object {
                "key": "Sundays",
                "value": 3,
              },
              Object {
                "key": "Thursdays",
                "value": 8,
              },
              Object {
                "key": "Tuesdays",
                "value": 2,
              },
              Object {
                "key": "Wednesdays",
                "value": 6,
              },
            ]
          }
          onClick={[Function]}
          x="key"
          y="value"
        />
      </FlexGridRow>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <BarChart
            data={
              Array [
                Object {
                  "key": "Branch A",
                  "value": 4,
                },
                Object {
                  "key": "Branch B",
                  "value": 4,
                },
                Object {
                  "key": "Branch C",
                  "value": 3,
                },
                Object {
                  "key": "Branch D",
                  "value": 6,
                },
                Object {
                  "key": "Branch E",
                  "value": 3,
                },
                Object {
                  "key": "Branch F",
                  "value": 7,
                },
              ]
            }
            onClick={[Function]}
            x="key"
            y="value"
          />,
          <BarChart
            data={
              Array [
                Object {
                  "key": "Rai",
                  "value": 2,
                },
                Object {
                  "key": "Reggai",
                  "value": 2,
                },
                Object {
                  "key": "Zahra",
                  "value": 2,
                },
                Object {
                  "key": "Qibla",
                  "value": 2,
                },
                Object {
                  "key": "Doha",
                  "value": 1,
                },
                Object {
                  "key": "Fahaheel",
                  "value": 1,
                },
                Object {
                  "key": "Abu Halifa",
                  "value": 1,
                },
                Object {
                  "key": "Jaber Al Ahmad",
                  "value": 1,
                },
                Object {
                  "key": "Al Qusour",
                  "value": 1,
                },
                Object {
                  "key": "Jaber Al Ali",
                  "value": 1,
                },
                Object {
                  "key": "Mirqab",
                  "value": 1,
                },
                Object {
                  "key": "Nahda",
                  "value": 1,
                },
                Object {
                  "key": "Nuzha",
                  "value": 1,
                },
                Object {
                  "key": "Qasser",
                  "value": 1,
                },
                Object {
                  "key": "Rabiya",
                  "value": 1,
                },
                Object {
                  "key": "Hadiya",
                  "value": 1,
                },
                Object {
                  "key": "Qortuba",
                  "value": 1,
                },
                Object {
                  "key": "Faiha",
                  "value": 1,
                },
                Object {
                  "key": "Al Bidae",
                  "value": 1,
                },
                Object {
                  "key": "Abu Al Hasaniya",
                  "value": 1,
                },
              ]
            }
            onClick={[Function]}
            x="key"
            y="value"
          />,
          <BarChart
            data={
              Array [
                Object {
                  "key": "Fridays",
                  "value": 1,
                },
                Object {
                  "key": "Mondays",
                  "value": 1,
                },
                Object {
                  "key": "Saturdays",
                  "value": 6,
                },
                Object {
                  "key": "Sundays",
                  "value": 3,
                },
                Object {
                  "key": "Thursdays",
                  "value": 8,
                },
                Object {
                  "key": "Tuesdays",
                  "value": 2,
                },
                Object {
                  "key": "Wednesdays",
                  "value": 6,
                },
              ]
            }
            onClick={[Function]}
            x="key"
            y="value"
          />,
        ],
        "gutter": 48,
        "justify": "space-around",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Array [
              Object {
                "key": "Branch A",
                "value": 4,
              },
              Object {
                "key": "Branch B",
                "value": 4,
              },
              Object {
                "key": "Branch C",
                "value": 3,
              },
              Object {
                "key": "Branch D",
                "value": 6,
              },
              Object {
                "key": "Branch E",
                "value": 3,
              },
              Object {
                "key": "Branch F",
                "value": 7,
              },
            ],
            "onClick": [Function],
            "x": "key",
            "y": "value",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Array [
              Object {
                "key": "Rai",
                "value": 2,
              },
              Object {
                "key": "Reggai",
                "value": 2,
              },
              Object {
                "key": "Zahra",
                "value": 2,
              },
              Object {
                "key": "Qibla",
                "value": 2,
              },
              Object {
                "key": "Doha",
                "value": 1,
              },
              Object {
                "key": "Fahaheel",
                "value": 1,
              },
              Object {
                "key": "Abu Halifa",
                "value": 1,
              },
              Object {
                "key": "Jaber Al Ahmad",
                "value": 1,
              },
              Object {
                "key": "Al Qusour",
                "value": 1,
              },
              Object {
                "key": "Jaber Al Ali",
                "value": 1,
              },
              Object {
                "key": "Mirqab",
                "value": 1,
              },
              Object {
                "key": "Nahda",
                "value": 1,
              },
              Object {
                "key": "Nuzha",
                "value": 1,
              },
              Object {
                "key": "Qasser",
                "value": 1,
              },
              Object {
                "key": "Rabiya",
                "value": 1,
              },
              Object {
                "key": "Hadiya",
                "value": 1,
              },
              Object {
                "key": "Qortuba",
                "value": 1,
              },
              Object {
                "key": "Faiha",
                "value": 1,
              },
              Object {
                "key": "Al Bidae",
                "value": 1,
              },
              Object {
                "key": "Abu Al Hasaniya",
                "value": 1,
              },
            ],
            "onClick": [Function],
            "x": "key",
            "y": "value",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Array [
              Object {
                "key": "Fridays",
                "value": 1,
              },
              Object {
                "key": "Mondays",
                "value": 1,
              },
              Object {
                "key": "Saturdays",
                "value": 6,
              },
              Object {
                "key": "Sundays",
                "value": 3,
              },
              Object {
                "key": "Thursdays",
                "value": 8,
              },
              Object {
                "key": "Tuesdays",
                "value": 2,
              },
              Object {
                "key": "Wednesdays",
                "value": 6,
              },
            ],
            "onClick": [Function],
            "x": "key",
            "y": "value",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <FlexGridRow
          gutter={48}
          justify="space-around"
        >
          <BarChart
            data={
              Array [
                Object {
                  "key": "Branch A",
                  "value": 4,
                },
                Object {
                  "key": "Branch B",
                  "value": 4,
                },
                Object {
                  "key": "Branch C",
                  "value": 3,
                },
                Object {
                  "key": "Branch D",
                  "value": 6,
                },
                Object {
                  "key": "Branch E",
                  "value": 3,
                },
                Object {
                  "key": "Branch F",
                  "value": 7,
                },
              ]
            }
            onClick={[Function]}
            x="key"
            y="value"
          />
          <BarChart
            data={
              Array [
                Object {
                  "key": "Rai",
                  "value": 2,
                },
                Object {
                  "key": "Reggai",
                  "value": 2,
                },
                Object {
                  "key": "Zahra",
                  "value": 2,
                },
                Object {
                  "key": "Qibla",
                  "value": 2,
                },
                Object {
                  "key": "Doha",
                  "value": 1,
                },
                Object {
                  "key": "Fahaheel",
                  "value": 1,
                },
                Object {
                  "key": "Abu Halifa",
                  "value": 1,
                },
                Object {
                  "key": "Jaber Al Ahmad",
                  "value": 1,
                },
                Object {
                  "key": "Al Qusour",
                  "value": 1,
                },
                Object {
                  "key": "Jaber Al Ali",
                  "value": 1,
                },
                Object {
                  "key": "Mirqab",
                  "value": 1,
                },
                Object {
                  "key": "Nahda",
                  "value": 1,
                },
                Object {
                  "key": "Nuzha",
                  "value": 1,
                },
                Object {
                  "key": "Qasser",
                  "value": 1,
                },
                Object {
                  "key": "Rabiya",
                  "value": 1,
                },
                Object {
                  "key": "Hadiya",
                  "value": 1,
                },
                Object {
                  "key": "Qortuba",
                  "value": 1,
                },
                Object {
                  "key": "Faiha",
                  "value": 1,
                },
                Object {
                  "key": "Al Bidae",
                  "value": 1,
                },
                Object {
                  "key": "Abu Al Hasaniya",
                  "value": 1,
                },
              ]
            }
            onClick={[Function]}
            x="key"
            y="value"
          />
          <BarChart
            data={
              Array [
                Object {
                  "key": "Fridays",
                  "value": 1,
                },
                Object {
                  "key": "Mondays",
                  "value": 1,
                },
                Object {
                  "key": "Saturdays",
                  "value": 6,
                },
                Object {
                  "key": "Sundays",
                  "value": 3,
                },
                Object {
                  "key": "Thursdays",
                  "value": 8,
                },
                Object {
                  "key": "Tuesdays",
                  "value": 2,
                },
                Object {
                  "key": "Wednesdays",
                  "value": 6,
                },
              ]
            }
            onClick={[Function]}
            x="key"
            y="value"
          />
        </FlexGridRow>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <BarChart
              data={
                Array [
                  Object {
                    "key": "Branch A",
                    "value": 4,
                  },
                  Object {
                    "key": "Branch B",
                    "value": 4,
                  },
                  Object {
                    "key": "Branch C",
                    "value": 3,
                  },
                  Object {
                    "key": "Branch D",
                    "value": 6,
                  },
                  Object {
                    "key": "Branch E",
                    "value": 3,
                  },
                  Object {
                    "key": "Branch F",
                    "value": 7,
                  },
                ]
              }
              onClick={[Function]}
              x="key"
              y="value"
            />,
            <BarChart
              data={
                Array [
                  Object {
                    "key": "Rai",
                    "value": 2,
                  },
                  Object {
                    "key": "Reggai",
                    "value": 2,
                  },
                  Object {
                    "key": "Zahra",
                    "value": 2,
                  },
                  Object {
                    "key": "Qibla",
                    "value": 2,
                  },
                  Object {
                    "key": "Doha",
                    "value": 1,
                  },
                  Object {
                    "key": "Fahaheel",
                    "value": 1,
                  },
                  Object {
                    "key": "Abu Halifa",
                    "value": 1,
                  },
                  Object {
                    "key": "Jaber Al Ahmad",
                    "value": 1,
                  },
                  Object {
                    "key": "Al Qusour",
                    "value": 1,
                  },
                  Object {
                    "key": "Jaber Al Ali",
                    "value": 1,
                  },
                  Object {
                    "key": "Mirqab",
                    "value": 1,
                  },
                  Object {
                    "key": "Nahda",
                    "value": 1,
                  },
                  Object {
                    "key": "Nuzha",
                    "value": 1,
                  },
                  Object {
                    "key": "Qasser",
                    "value": 1,
                  },
                  Object {
                    "key": "Rabiya",
                    "value": 1,
                  },
                  Object {
                    "key": "Hadiya",
                    "value": 1,
                  },
                  Object {
                    "key": "Qortuba",
                    "value": 1,
                  },
                  Object {
                    "key": "Faiha",
                    "value": 1,
                  },
                  Object {
                    "key": "Al Bidae",
                    "value": 1,
                  },
                  Object {
                    "key": "Abu Al Hasaniya",
                    "value": 1,
                  },
                ]
              }
              onClick={[Function]}
              x="key"
              y="value"
            />,
            <BarChart
              data={
                Array [
                  Object {
                    "key": "Fridays",
                    "value": 1,
                  },
                  Object {
                    "key": "Mondays",
                    "value": 1,
                  },
                  Object {
                    "key": "Saturdays",
                    "value": 6,
                  },
                  Object {
                    "key": "Sundays",
                    "value": 3,
                  },
                  Object {
                    "key": "Thursdays",
                    "value": 8,
                  },
                  Object {
                    "key": "Tuesdays",
                    "value": 2,
                  },
                  Object {
                    "key": "Wednesdays",
                    "value": 6,
                  },
                ]
              }
              onClick={[Function]}
              x="key"
              y="value"
            />,
          ],
          "gutter": 48,
          "justify": "space-around",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "data": Array [
                Object {
                  "key": "Branch A",
                  "value": 4,
                },
                Object {
                  "key": "Branch B",
                  "value": 4,
                },
                Object {
                  "key": "Branch C",
                  "value": 3,
                },
                Object {
                  "key": "Branch D",
                  "value": 6,
                },
                Object {
                  "key": "Branch E",
                  "value": 3,
                },
                Object {
                  "key": "Branch F",
                  "value": 7,
                },
              ],
              "onClick": [Function],
              "x": "key",
              "y": "value",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "data": Array [
                Object {
                  "key": "Rai",
                  "value": 2,
                },
                Object {
                  "key": "Reggai",
                  "value": 2,
                },
                Object {
                  "key": "Zahra",
                  "value": 2,
                },
                Object {
                  "key": "Qibla",
                  "value": 2,
                },
                Object {
                  "key": "Doha",
                  "value": 1,
                },
                Object {
                  "key": "Fahaheel",
                  "value": 1,
                },
                Object {
                  "key": "Abu Halifa",
                  "value": 1,
                },
                Object {
                  "key": "Jaber Al Ahmad",
                  "value": 1,
                },
                Object {
                  "key": "Al Qusour",
                  "value": 1,
                },
                Object {
                  "key": "Jaber Al Ali",
                  "value": 1,
                },
                Object {
                  "key": "Mirqab",
                  "value": 1,
                },
                Object {
                  "key": "Nahda",
                  "value": 1,
                },
                Object {
                  "key": "Nuzha",
                  "value": 1,
                },
                Object {
                  "key": "Qasser",
                  "value": 1,
                },
                Object {
                  "key": "Rabiya",
                  "value": 1,
                },
                Object {
                  "key": "Hadiya",
                  "value": 1,
                },
                Object {
                  "key": "Qortuba",
                  "value": 1,
                },
                Object {
                  "key": "Faiha",
                  "value": 1,
                },
                Object {
                  "key": "Al Bidae",
                  "value": 1,
                },
                Object {
                  "key": "Abu Al Hasaniya",
                  "value": 1,
                },
              ],
              "onClick": [Function],
              "x": "key",
              "y": "value",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "data": Array [
                Object {
                  "key": "Fridays",
                  "value": 1,
                },
                Object {
                  "key": "Mondays",
                  "value": 1,
                },
                Object {
                  "key": "Saturdays",
                  "value": 6,
                },
                Object {
                  "key": "Sundays",
                  "value": 3,
                },
                Object {
                  "key": "Thursdays",
                  "value": 8,
                },
                Object {
                  "key": "Tuesdays",
                  "value": 2,
                },
                Object {
                  "key": "Wednesdays",
                  "value": 6,
                },
              ],
              "onClick": [Function],
              "x": "key",
              "y": "value",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
